### PR TITLE
Heat equation with two layers

### PR DESCRIPTION
**Pull request by bmxam**
This is the adaptation of a legacy-API example. Only a small part has been adapted, primarily to demonstrate the use of subdomain integration.

Need to wait for Bcube to remove the assert preventing the subdomain integration on non-dense indexing.

### CODE REVIEW

**Comment by bmxam** [comment]
Bcube is ready but it would require a new tag. Let's wait for 0.1.8

File: src/example/heat_equation_two_layers/heat_equation_two_layers.jl [hunk 1]
```diff
@@ -0,0 +1,168 @@
+module HeatEquationTwoLayers #hide
+println("Running heat equation two layers example...") #hide
+# # Heat equation
+# # Theory
+# This example shows how to solve the heat equation with eventually variable physical properties in steady and unsteady formulations:
+# ```math
+#   \rho C_p \partial_t u - \nabla . ( \lambda u) = f
```
Comment (lokmanb) [inline src/example/heat_equation_two_layers/heat_equation_two_layers.jl:7]: Small typo (probably coming from the legacy file), there is a \nabla missing before u.

**Review by lokmanb — APPROVED** [review]
Nice illustration of the subdomain integration feature!

File: src/example/heat_equation_two_layers/heat_equation_two_layers.jl (hunk 1 above)
Comment (bmxam) [reply src/example/heat_equation_two_layers/heat_equation_two_layers.jl]: fixed

**Pull request merged**